enhancements/storage/vsphere-driver-configuration.md

```diff
@@ -0,0 +1,286 @@
+3. The driver controller pods are restarted.
+4. The cluster administrator creates a VolumeSnapshot object and the driver allows to create up to 5 snapshots per volume.
+
+Option 2: Configuration via generic map in a dedicated field of ClusterCSIDriver
+
+1. The cluster administrator creates a ClusterCSIDriver object with the configuration map set to `{"maxSnapshots": "5"}`
+under a generic `map[string]string` field (`clustercsidriver.spec.vSphere.configuration`)
+2. The driver operator reconciles the ClusterCSIDriver object and propagates the option to a valid destination (ini file or environment variable in case of snapshot limit)
+3. The driver controller pods are restarted.
+4. The cluster administrator creates a VolumeSnapshot object and the driver allows to create up to 5 snapshots per volume.
+
+
+Option 3: Mirror configuration in ClusterCSIDriver as fields (generic)
+
+1. The cluster administrator creates a ClusterCSIDriver object with the maximum limit of snapshots set to 10 using a
+   specific integer field (`clustercsidriver.spec.driverConfig.vSphereConfig.GlobalMaxSnapshotsPerBlockVolume`).
+2. Optionally cluster administrator can add any other option documented by CSI driver, for example decrease snapshot
+limit for VSAN to 5 (`clustercsidriver.spec.driverConfig.vSphereConfig.GranularMaxSnapshotsPerBlockVolumeInVSAN`)
+3. The driver operator reconciles the ClusterCSIDriver object and propagates the option to a valid destination (ini file or environment variable in case of snapshot limit)
+4. The driver controller pods are restarted.
+5. The cluster administrator creates a VolumeSnapshot object and the driver allows to create up to 10 snapshots per volume and 5 snapshots per volume in VSAN.
+
+#### Variation and form factor considerations [optional]
+
+None.
+
+### API Extensions
+
+All options considered require changes in the ClusterCSIDriver API so that a new field or fields are added to the
+spec. No other changes or new CRDs are required.
+
+### Implementation Details/Notes/Constraints [optional]
+
+None. See the [Proposal](#Proposal) for details.
+
+#### Hypershift [optional]
+
+No specifics are required for Hypershift - users already have access to ClusterCSIDriver CR in guest cluster and can modify it.
+
+### Risks and Mitigations
+
+Considerate risk would be allowing users to use any configuration option. This would lead to increased testing complexity
+and potential for more support tickets. We can mitigate this by releasing the feature as TechPreview and gather feedback
+as described here: https://github.com/openshift/enhancements/blob/2894d936eca7c1fcc9cf38e7dc973bbdfa1d88ff/dev-guide/api-conventions.md#new-field-added-to-an-existing-crd
+
+On the other hand if we take the safe route and implement each option separately as new fields in the CRD, we're risking
+a scenario where an option or feature could be suddenly dropped upstream, and we can not simply remove fields from release
+to release.
```

Comment:
We have this problem in each of the options described above, as it is somewhat orthogonal to *how* we expose configuration options. Any CSI driver could remove any feature / config option at any time

It would be nice to also mention that CSI drivers have much less rigorous feature development and removal process than k/k.

Reply:
Good point, I rewrote this section to emphasize this.